adafruit_tmp006.py

```diff
@@ -0,0 +1,182 @@
+CFG_16SAMPLE = const(0x0800)
+_TMP006_CFG_DRDYEN = const(0x0100)
+_TMP006_CFG_DRDY = const(0x0080)
+
+class TMP006:
+    """Class to represent an Adafruit TMP006 non-contact temperature measurement
+    board.
+    """
+
+    # Class-level buffer for reading and writing data with the sensor.
+    # This reduces memory allocations but means the code is not re-entrant or
+    # thread safe!
+    _BUFFER = bytearray(4)
+
+    def __init__(self, i2c, address=_TMP006_I2CADDR, samplerate=CFG_16SAMPLE):
+        self._device = I2CDevice(i2c, address)
+        self._write_u16(_TMP006_CONFIG, _TMP006_CFG_RESET)
+        time.sleep(.5)
+        if samplerate not in (CFG_1SAMPLE, CFG_2SAMPLE, CFG_4SAMPLE, CFG_8SAMPLE,
+                              CFG_16SAMPLE):
+            raise ValueError('Unexpected samplerate value! Must be one of: ' \
+                'CFG_1SAMPLE, CFG_2SAMPLE, CFG_4SAMPLE, CFG_8SAMPLE, or CFG_16SAMPLE')
+        # Set configuration register to turn on chip, enable data ready output,
+        # and start sampling at the specified rate.
+        config = _TMP006_CFG_MODEON | _TMP006_CFG_DRDYEN | samplerate
+        self._write_u16(_TMP006_CONFIG, config)
+        # Check device ID match expected value.
+        dev_id = self.read_register(_TMP006_DEVID)
+        if dev_id != 0x67:
+            raise RuntimeError('Init failed - Did not find TMP006')
+
+    @property
+    def active(self):
+        """True if sensor is active."""
+        return self._read_u16(_TMP006_CONFIG) & _TMP006_CFG_MODEON != 0
+
+    @active.setter
+    def active(self, val):
+        if val:
+            self._wake()
+        else:
+            self._sleep()
+
+    @property
+    def temperature(self):
+        # pylint: disable=bad-whitespace, invalid-name, too-many-locals
+        """Read object temperature from TMP006 sensor."""
+        if not self.active:
+            raise RuntimeError('Can not read from sensor when inactive.')
+        while not self._data_ready():
+            pass
+        vobj = self.read_register(_TMP006_VOBJ)
+        vobj = bytearray((vobj & 0xFF, vobj >> 8 & 0xFF))
+        vobj = struct.unpack("<h", vobj)[0]
+        vobj = vobj * 156.25e-9  # volts
+        tamb = self.read_register(_TMP006_TAMB)
+        tamb = bytearray((tamb & 0xFF, tamb >> 8 & 0xFF))
+        tamb = struct.unpack("<h", tamb)[0]
+        tamb = (tamb >> 2) / 32.
+        tamb += 273.15 # kelvin
+        # see TMP006 User Guide, section 5.1
+        S0 = 6.4e-14 # nominal value
+        a1 = 1.75e-3
+        a2 = -1.678e-5
+        TREF = 298.15
+        b0 = -2.94e-5
+        b1 = -5.7e-7
+        b2 = 4.63e-9
+        c2 = 13.4
+
+        S = S0 * (1 + a1*(tamb - TREF) + a2*(tamb - TREF)**2)
+        VOS = b0 + b1*(tamb - TREF) + b2*(tamb - TREF)**2
+        fVOBJ = (vobj - VOS) + c2*(vobj - VOS)**2
+
+        TOBJ = (tamb**4 + (fVOBJ/S))**0.25
+
+        TOBJ -= 273.15 # back to celsius
+
+        return TOBJ
+
+    def _sleep(self):
```

Comment:
_sleep and _wake are basically the same function, combine the code and put into the active.setter?

Reply:
good call. i was just lazily reusing the tmp007 funcs.